src/authorization/WebAclAuthorizer.ts

```diff
@@ -184,20 +184,25 @@ export class WebAclAuthorizer extends Authorizer {
    *
    * @returns A store containing the relevant acl triples.
    */
-  private async getAclRecursive(id: ResourceIdentifier, recurse?: boolean): Promise<Store> {
+  private async getAclRecursive(id: ResourceIdentifier, originalId?: ResourceIdentifier): Promise<Store> {
+    const target = originalId ?? id;
     this.logger.debug(`Trying to read the direct ACL document of ${id.path}`);
     try {
       const acl = this.aclStrategy.getAuxiliaryIdentifier(id);
       this.logger.debug(`Trying to read the ACL document ${acl.path}`);
       const data = await this.resourceStore.getRepresentation(acl, { type: { [INTERNAL_QUADS]: 1 }});
       this.logger.info(`Reading ACL statements from ${acl.path}`);
+      const paths: [ predicate: string, object: string ][] = [
+        [ ACL.accessTo, target.path ],
+        [ ACL.default, id.path ],
+      ];
```

Comment:
This would change the behaviour of the `getAclRecursive` function (unless I'm misreading?). Specifically, when accessing `/foo/`, if `/foo/.acl` contains `... acl:default <./>` this would now allow the operation where previously it would not.

When I mentioned moving the functionality to `filterData` I was also thinking of moving both these predicates there as well. `filterData` could then take 2 identifiers as input, with one of them being optional and determining if `acl:default` also needs to be checked.

Reply:
If `filterData` becomes so specific, then why is it a seperate function? That's also why I asked whether the function shouldn't be in some utility module.

Reply:
To separate responsibilities for functions making the code more readable. Most functions in this class have very specific jobs and are only called in 1 specific place (e.g., `checkPermission`, `hasPermission`, etc.) but putting all that in a single function would make this class a lot less readable.

Specifically here, the main focus of `getAclRecursive` is finding the correct acl resource that has information for the given identifier. Parsing and extracting information from such an acl resource is not part of the main goal of this function so is delegated to a different function to not muddy up `getAclRecursive`.

This is all partially subjective of course.

Reply:
Thanks for spotting that regression though!

Reply:
Of course I know about separation of concerns, single responsibility etc., but this typically also entails that a lot of separated functionality becomes shareable, something which I definitely think was the case with the `filterData` function. Anyhow, I renamed the function so the specificity is more clear, fixed the regression and added a test for it.

Reply:
You're not wrong of course that this does make the function more specific. But since it's not used anywhere else I don't see this as a big concern since it still does this one specific thing. It's just slightly changed. Should we ever need similar behaviour elsewhere we could look into extracting some parts to a separate utility library.